### PR TITLE
ROX-13582: Gathering central datastore properties

### PR DESCRIPTION
**Pull request by 0x656b694d**
## Description

A piece of #3819.

Periodic data stores gathering for phone-home telemetry purposes.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Unit tests.

Example central identity message, received by Segment:
```json
{
  "context": {
    "Client ID": "f5179232-8352-404a-b2e2-fa5efcaccca5",
    "Client Name": "Central",
    "library": {
      "name": "analytics-go",
      "version": "3.0.0"
    }
  },
  "integrations": {},
  "messageId": "e1e990b3-5361-4353-9b80-3ccacaee0655",
  "originalTimestamp": "2022-12-08T17:28:53.833816386Z",
  "receivedAt": "2022-12-08T17:29:53.988Z",
  "sentAt": "2022-12-08T17:29:53.305Z",
  "timestamp": "2022-12-08T17:28:54.516Z",
  "traits": {
    "Auth Providers": [
      "saml"
    ],
    "Central version": "3.73.x-156-g65733eecef-dirty",
    "Chart version": "73.0.156-g65733eecef-dirty",
    "Kubernetes version": "v1.25.3",
    "Orchestrator": "KUBERNETES_CLUSTER",
    "Total Access Scopes": 2,
    "Total Groups of saml": 1,
    "Total PermissionSets": 9,
    "Total Roles": 9,
    "Total Signature Integrations": 0
  },
  "type": "identify",
  "userId": "f5179232-8352-404a-b2e2-fa5efcaccca5",
  "writeKey": "..."
}
```

### CODE REVIEW

**Comment by 0x656b694d** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4070** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4070" target="_blank">Graphite</a> 
    * **PR #4071** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4071" target="_blank">Graphite</a>  👈
      * **PR #4072** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4072" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/4071?utm_source=stack-comment).

File: pkg/telemetry/phonehome/utils.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
```
Comment (dhaus67) [inline pkg/telemetry/phonehome/utils.go:12]: nit: doesn't seem to be used within this file but rather `client_config.go / gatherer.go`, so might make sense to move it closer to where it's actually used.

**Review by dhaus67 — APPROVED** [review]
Minor comments, otherwise LGTM 🚢

File: pkg/telemetry/phonehome/utils.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package phonehome
```
Comment (stehessel) [inline pkg/telemetry/phonehome/utils.go:1]: I think "utils" files are best avoided as they are too generic. Maybe you can move this function somewhere else and delete this file all together?

**Review by stehessel — APPROVED** [review]

File: central/authprovider/datastore/telemetry.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	ctx = sac.WithAllAccess(ctx)
+	props := make(map[string]any)
+
+	providers, err := Singleton().GetAllAuthProviders(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get AuthProviders")
+	}
+
+	providerIDTypes := make(map[string]string, len(providers))
+	providerTypes := set.NewSet[string]()
```
[the remark this answers is not in the document]
Comment (0x656b694d) [reply central/authprovider/datastore/telemetry.go:26]: I left some of the `New*Set` aliases to reduce the code diff when moving to generics. Not sure if using those aliases for the new code is what we're going to prefer.

File: central/main.go [hunk 4]
```diff
@@ -312,6 +314,7 @@ func startServices() {
 	pruning.Singleton().Start()
 	gatherer.Singleton().Start()
 	vulnRequestManager.Singleton().Start()
+	centralclient.InstanceConfig().Gatherer().Start()
```
[the remark this answers is not in the document]
Comment (0x656b694d) [reply central/main.go:317]: I don't see adding import names for the case camelling only (see `authproviders`, `clientconn`, `servicecerttoken`). We only add names to resolve conflicts or for cutting the names short.

File: go.mod [hunk 5]
```diff
@@ -359,6 +360,11 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 )
 
+require (
+	github.com/segmentio/backo-go v1.0.1 // indirect
+	github.com/xtgo/uuid v0.0.0-20140804021211-a0b114877d4c // indirect
+)
+
```
[the remark this answers is not in the document]
Comment (0x656b694d) [reply go.mod:367]: Hm, there's something that makes vscode creating this section. I'll merge, thanks.

**Pull request merged**